alica_engine/src/engine/modelmanagement/factories/Factory.cpp

```diff
@@ -116,7 +116,7 @@ void Factory::storeElement(AlicaElement* ael, const std::string& type)
         ss << "ELEMENT >" << ael->getName() << "< >" << modelManager->elements[ael->getId()]->getName() << "<" << std::endl;
         AlicaEngine::abort(ss.str());
     }
-    modelManager->elements.insert(std::pair<int64_t, AlicaElement*>(ael->getId(), ael));
+    modelManager->elements.insert(std::pair<int64_t, std::unique_ptr<AlicaElement>>(ael->getId(), ael));
```

Comment:
although this does fix the memory leak, I don't think it is the cause of continuous memory increase during the running of lbc, because the models are parsed & stored once & their lifetime is equal to the lifetime of the engine.

Reply:
It significantly reduces the size of valgrind's log.